urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/ui/simple/SimpleTheme.java

```diff
@@ -0,0 +1,68 @@
+package com.kokakiwi.mclauncher.ui.simple;
+
+import java.awt.Dimension;
+import java.awt.Image;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.imageio.ImageIO;
+
+import com.kokakiwi.mclauncher.api.LauncherAPI;
+import com.kokakiwi.mclauncher.api.ui.Page;
+import com.kokakiwi.mclauncher.api.ui.Theme;
+
+public class SimpleTheme implements Theme
+{
+    private LauncherAPI       api;
+    
+    private SimpleLoginPage   loginPage;
+    private SimpleLoadingPage loadingPage;
+    
+    private Image             background = null;
+    
+    public void onLoad(LauncherAPI api)
+    {
+        this.api = api;
+        
+        loginPage = new SimpleLoginPage();
+        loadingPage = new SimpleLoadingPage();
+    }
+    
+    public List<Page> getPages()
+    {
+        final List<Page> pages = new ArrayList<Page>();
+        
+        pages.add(loginPage);
+        pages.add(loadingPage);
+        
+        return pages;
+    }
+    
+    public Dimension getDimension()
+    {
+        return new Dimension(854, 480);
+    }
+    
+    public Image getBackground()
+    {
+        if (background == null)
+        {
+            try
+            {
+                background = ImageIO.read(
+                        SimpleTheme.class.getResourceAsStream("/"
+                                + api.getConfig().getString(
+                                        "theme.simple.background")))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getResourceAsStream*. Currently, there are execution paths that do not contain closure statements, for example, when ImageIO.read throws an exception. Ensure the object returned by *getResourceAsStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).